Update Daml Shell documentation for where-queries (#653)
digital-asset/docs.daml.com: docs/2.9.0/docs/tools/daml-shell/index.rst

     --identifier-trim-location <leading|middle|trailing>
                              Where to trim long identifiers. Default: "trailing"
     --disable-color          Disable ANSI colored output.

>      --postgres-host <host>   Postgres host to connect to. Implies --connect. Default: "localhost"

Author: That's a good point... Unfortunately this mirrors the naming scheme of PQS/Scribe, which means that this change is even more entrenched by now.

The auto-completion provides both FQN variants (with and without package
ID).

Filtering with ``where`` clauses
================================

To refine your queries when listing contracts, you can use ``where``
clauses to filter based on specific payload fields. This uses a SQL-like
syntax for conditionals, and is supported for the ``active``,
``creates``, ``archives`` and ``exercises`` commands.

> Nested field names are identified using dot notation such as
> ``parent.child.value``.

Reviewer: ```suggestion
To nest fields, use dot notation: ``parent.child.value``
```

Supported comparison operators

> ==============================

Reviewer: From here down, I suggest demoting the headings to h4. Works better within the whole page:
<img width="345" alt="Screenshot 2024-02-28 at 5 52 14 PM" src="https://github.com/digital-asset/docs.daml.com/assets/139921049/1242a4f0-8f53-4255-95c3-27e18b275f1d">

> -  ``=``: Equal to

Reviewer: I think it's more readable without the colons after the operators, but your call.

(but keep the colons for the Logical operators, because there they aid readability) 🙃

-  ``!=``: Not equal to
-  ``>``: Greater than
-  ``>=``: Greater than or equal to
-  ``<``: Less than
-  ``<=``: Less than or equal to
-  ``like``: Used for pattern matching, ``%`` serves as a wildcard
   character

> Logical operators
> =================

Reviewer: ```suggestion
Logical operators
-----------------
```

-  ``and``: Both conditions must be satisfied
-  ``or``: Either condition may be satisfied

You can use parentheses to group conditions and direct the order of
evaluation.

Type casting in ``where`` clauses
=================================

To ensure proper comparison, you can optionally cast fields to a
specific type using the ``::`` operator. Supported types for casting
are: - ``numeric``: For numerical comparisons - ``timestamp``: For

> date/time comparisons - ``text``: For textual comparisons

Author: Thanks, I removed the redundant list.

Field values are sorted and compared lexicographically if no cast is
specified.

``where`` clause examples
=========================

> Below are examples of how to use ``where`` clauses in commands:

Reviewer: ```suggestion
Here are some examples of how to use ``where`` clauses in commands:
```

-  Filter by a string pattern:

   ::

      > active where owner like Alice%

   Lists contracts where the ``owner`` field starts with the string
   ``Alice``.

-  Filter by a nested numeric field:

   ::

      > active where deeply.nested.value :: numeric > 1000

>    Lists contracts where a the nested field ``value`` is greater than

Reviewer: ```suggestion
   Lists contracts where the nested field ``value`` is greater than
```

   ``1000``.

-  Filter with exact string match (note the use of double quotes):

   ::

      > active where label = "loren ipsum"

   Lists contracts where the label field is exactly ``loren ipsum``.

>    Double quotes allow values to contain whitespaces.

Reviewer: ```suggestion
   Use double quotes with values that contain whitespace characters.
```
